### ISSUE TITLE
Logic of lr_scheduler.step() using DeepSpeed and multi-gpu

### ISSUE THREAD

**Issue by memray** [huggingface/accelerate#1096]
### System Info

```Shell
- `Accelerate` version: 0.16.0
- Platform: Linux-5.10.133+-x86_64-with-glibc2.17
- Python version: 3.8.12
- Numpy version: 1.22.2
- PyTorch version (GPU?): 1.11.0+cu113 (True)
- `Accelerate` default config:
	- compute_environment: LOCAL_MACHINE
	- distributed_type: DEEPSPEED
	- mixed_precision: bf16
	- use_cpu: False
	- dynamo_backend: NO
	- num_processes: 16
	- machine_rank: 0
	- num_machines: 1
	- rdzv_backend: static
	- same_network: True
	- main_training_function: main
	- deepspeed_config: {'gradient_accumulation_steps': 4, 'gradient_clipping': 1.0, 'offload_optimizer_device': 'cpu', 'offload_param_device': 'cpu', 'zero3_init_flag': True, 'zero_stage': 2}
	- fsdp_config: {}
	- megatron_lm_config: {}
	- downcast_bf16: no
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

Code: https://github.com/huggingface/accelerate/blob/main/examples/by_feature/deepspeed_with_config_support.py

Add a line of printing at [L650](https://github.com/huggingface/accelerate/blob/ea36b7dcebd5292afba328291a2ec6361dbcfb9a/examples/by_feature/deepspeed_with_config_support.py#L650): `logger.info(f"\nTRAIN Step {completed_steps}: train_loss: {loss.detach().float()}, lr_step: {lr_scheduler.scheduler._step_count}, lr: {lr_scheduler.get_last_lr()[0]}", main_process_only=False)`

Command:
`CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15 accelerate launch --config_file gpu16-Stage-2-bf16-cpu-config.yaml deepspeed_with_config_support.py --model_name_or_path salesforce/codegen-2B-multi --dataset_name wikitext --dataset_config_name wikitext-103-v1 --per_device_train_batch_size 1 --per_device_eval_batch_size 2 --output_dir output/FT-mktcloud-test --max_train_steps 20 --num_warmup_steps 5 --with_tracking --learning_rate 1e-5`

In this example, `lr_step` moves 16 times faster than `completed_steps`.
Since `lr_scheduler.split_batches` is `False` (default), the for loop in [scheduler.py#L73](https://github.com/huggingface/accelerate/blob/ea36b7dcebd5292afba328291a2ec6361dbcfb9a/src/accelerate/scheduler.py#L73) will be called `num_processes` times. And lr_scheduler will quickly reach the max_train_steps (in this example 20).
Is this intentionally designed?

Current output:
```Shell
TRAIN Step 0: train_loss: 0.9375, lr_step: 1, lr: 0.0
TRAIN Step 1: train_loss: 0.9375, lr_step: 17, lr: 2.666666666666667e-06
TRAIN Step 2: train_loss: 1.0546875, lr_step: 33, lr: 0.0
TRAIN Step 3: train_loss: 0.8671875, lr_step: 49, lr: 0.0
```

### Expected behavior
```Shell
Expected behavior:
TRAIN Step 0: train_loss: xxxx, lr_step: 1, lr: 0.0
TRAIN Step 1: train_loss: xxxx, lr_step: 2, lr: >0
TRAIN Step 2: train_loss: xxxx, lr_step: 3, lr: >0
TRAIN Step 3: train_loss: xxxx, lr_step: 4, lr: >0
```

**Comment by pacman100** [comment]
Hello @memray , could you please refer to https://github.com/huggingface/accelerate/issues/1095#issuecomment-1437923862 and let us know if that resolves the query you have?

**Comment by memray** [comment]
Hi @pacman100 , I was using `--lr_scheduler_type linear`, so the lr is controlled in the training loop (not by DeepSpeed). 

You mentioned 
> However, for AcceleratedScheduler doesn't reduce the max_train_steps to 125 and keeps it as 1000. So, after first call to accelerated_scheduler.step() it will result into lr being 4.96e-5 (i.e., 5e-5 * (1-8/1000)).

In this case, what hyperparameters should I use with this official example? Set the `--max_train_steps` to be `num_processes*wanted_train_steps`?

Also can you clarify why max_train_steps `needs to be divided by num_processes to account for data parallel`? What I understand by using multiple GPUs, it only affects the batch size per step. But why it affects training length here?

Thanks!

**Comment by muellerzr** [comment]
@memray this is because when you prepare a dataloader, `Accelerate` will assume the batch size is the number of items *per GPU*. So e.g. say you ran a script with a bs of 16 on a single GPU, launching on multiple GPUs (say 2) takes that to 32 total items looked at per processes, each GPU sees a batch size of 16. (This is then also why we step the scheduler/optimizer `n` times as well)
